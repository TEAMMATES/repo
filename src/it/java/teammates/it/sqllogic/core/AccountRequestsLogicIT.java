package teammates.it.sqllogic.core;

import java.time.Instant;
import java.util.UUID;

import org.testng.annotations.Test;

import teammates.common.datatransfer.AccountRequestStatus;
import teammates.common.exception.EntityAlreadyExistsException;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.AccountRequestsLogic;
import teammates.storage.sqlapi.AccountRequestsDb;
import teammates.storage.sqlentity.AccountRequest;

/**
 * SUT: {@link AccountRequestsLogic}.
 */
public class AccountRequestsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private AccountRequestsLogic accountRequestsLogic = AccountRequestsLogic.inst();

    @Test
    public void testGetAccountRequest_nonExistentAccountRequest_returnsNull() {
        UUID id = UUID.randomUUID();
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertNull(actualAccountRequest);
    }

    @Test
    public void testGetAccountRequest_existingAccountRequest_getsSuccessfully() throws InvalidParametersException {
        AccountRequest expectedAccountRequest =
                new AccountRequest("test@gmail.com", "name", "institute", AccountRequestStatus.PENDING, "comments");
        UUID id = expectedAccountRequest.getId();
        accountRequestsLogic.createAccountRequest(expectedAccountRequest);
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertEquals(expectedAccountRequest, actualAccountRequest);
    }

    @Test
    public void testResetAccountRequest()
            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {

        ______TS("success: create account request and update registeredAt field");

        String name = "name lee";
        String email = "email@gmail.com";
        String institute = "institute";
        AccountRequestStatus status = AccountRequestStatus.PENDING;
        String comments = "comments";

        AccountRequest toReset = accountRequestsLogic.createAccountRequest(name, email, institute, status, comments);
        AccountRequestsDb accountRequestsDb = AccountRequestsDb.inst();

        toReset.setRegisteredAt(Instant.now());
        toReset = accountRequestsDb.getAccountRequest(email, institute);

        assertNotNull(toReset);
        assertNotNull(toReset.getRegisteredAt());

        ______TS("success: reset account request that already exists");

        AccountRequest resetted = accountRequestsLogic.resetAccountRequest(email, institute);

        assertNull(resetted.getRegisteredAt());

        ______TS("success: test delete account request");

        accountRequestsLogic.deleteAccountRequest(email, institute);

        assertNull(accountRequestsLogic.getAccountRequest(email, institute));

        ______TS("failure: reset account request that does not exist");

        assertThrows(EntityDoesNotExistException.class,
                () -> accountRequestsLogic.resetAccountRequest(name, institute));
    }
}
