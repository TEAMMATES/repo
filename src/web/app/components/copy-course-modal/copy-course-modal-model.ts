import {FeedbackSession} from "../../../types/api-output";

/**
 * Result of {@link CopyCourseModalComponent}
 */
export interface CopyCourseModalResult {
    newCourseId: string;
    newCourseName: string;
    newTimeZone: string;
    chosenFeedbackSessionList: Set<FeedbackSession>;
}