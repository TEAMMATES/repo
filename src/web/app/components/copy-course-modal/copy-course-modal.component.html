<div class="modal-header">
    <h5 class="modal-title">
        <div>Course to copy: <b>{{oldCourseId}}</b></div>
    </h5>
    <button type="button" class="close" (click)="activeModal.dismiss()">
        <i class="fas fa-times"></i>
    </button>
</div>
<div id="copy-course-modal" class="modal-body">
    <div class="row">
        <div class="col-12">
            <div class="form-group">
                <label><b>New Course ID</b></label>
                <input id="copy-course-id" type="text" class="form-control" [(ngModel)]="newCourseId" [maxlength]="COURSE_ID_MAX_LENGTH">
                <span>{{ COURSE_ID_MAX_LENGTH - newCourseId.length }} characters left</span>
            </div>
            <div class="form-group">
                <label><b>Course Name</b></label>
                <input id="new-course-name" class="form-control" type="text" [(ngModel)]="newCourseName"
                       [maxlength]="64"/>
                <span>{{ 64 - newCourseName.length }} characters left</span>
            </div>
            <div class="form-group">
                <label><span ngbTooltip="You should not
                        need to change this as it is auto-detected based on your device settings. TEAMMATES automatically adjusts
                        to match the current time offset in your area, including clock changes due to daylight saving time.">
                <b>Time Zone</b></span></label>
                <div class="input-group">
                    <select id="new-time-zone" class="form-control" [(ngModel)]="newTimezone">
                        <option *ngFor="let timezone of timezones" [value]="timezone.id">{{ timezone.id }} ({{ timezone.offset }})</option>
                    </select>
                    <span class="input-group-btn" style="margin-left: 5px;">
                        <button class="btn btn-primary btn-margin-left" (click)="onAutoDetectTimezone()">Auto Detect</button>
                    </span>
                </div>
            </div>
            <div> <label><b>Select Feedback Sessions</b></label>
                <div class="form-check">
                    <input type="checkbox" name="copyCourseChooseSessions" class="form-check-input"
                           [checked]="chosenFeedbackSessions.size == sessionsInCourse.length" (click)="toggleSelection()">
                    <span>Select all sessions</span>
                </div>
            </div>
            <div class="form-check" *ngFor="let session of sessionsInCourse">
                <label class="form-check-label">
                    <input type="checkbox" name="copyCourseChooseSessions" class="form-check-input"
                           [checked]="chosenFeedbackSessions.has(session)" (click)="select(session)">
                    <span>{{ session.feedbackSessionName }}</span>
                </label>
            </div>
        </div>
    </div>
</div>
<div class="modal-footer">
    <button type="button" class="btn btn-light" (click)="activeModal.dismiss()">Cancel</button>
    <button id="btn-confirm-copy-course" type="button" class="btn btn-primary" (click)="copy()"
            [disabled]="!newCourseId || !newCourseName">Copy</button>
</div>
