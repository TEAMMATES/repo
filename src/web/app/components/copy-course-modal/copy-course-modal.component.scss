.modal-selection * {
  margin-right: .75rem;
  padding: .1rem .5rem;
}