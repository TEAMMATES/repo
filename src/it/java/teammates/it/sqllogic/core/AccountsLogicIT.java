package teammates.it.sqllogic.core;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

import org.testng.annotations.Test;

import teammates.common.datatransfer.NotificationStyle;
import teammates.common.datatransfer.NotificationTargetUser;
import teammates.common.exception.EntityAlreadyExistsException;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.AccountsLogic;
import teammates.sqllogic.core.NotificationsLogic;
import teammates.storage.sqlapi.AccountsDb;
import teammates.storage.sqlentity.Account;
import teammates.storage.sqlentity.Notification;
import teammates.storage.sqlentity.ReadNotification;

/**
 * SUT: {@link AccountsLogic}.
 */
public class AccountsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private AccountsLogic accountsLogic = AccountsLogic.inst();
    private NotificationsLogic notificationsLogic = NotificationsLogic.inst();

    private AccountsDb accountsDb = AccountsDb.inst();

    @Test
    public void testUpdateReadNotifications()
            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {
        ______TS("success: mark notification as read");
        Account account = new Account("google-id", "name", "email@teammates.com");
        Notification notification = new Notification(Instant.parse("2011-01-01T00:00:00Z"),
                Instant.parse("2099-01-01T00:00:00Z"), NotificationStyle.DANGER, NotificationTargetUser.GENERAL,
                "A deprecation note", "<p>Deprecation happens in three minutes</p>");
        accountsDb.createAccount(account);
        notificationsLogic.createNotification(notification);

        String googleId = account.getGoogleId();
        UUID notificationId = notification.getId();
        accountsLogic.updateReadNotifications(googleId, notificationId, notification.getEndTime());

        Account actualAccount = accountsDb.getAccountByGoogleId(googleId);
        List<ReadNotification> accountReadNotifications = actualAccount.getReadNotifications();
        assertEquals(1, accountReadNotifications.size());
        assertSame(actualAccount, accountReadNotifications.get(0).getAccount());
        assertSame(notification, accountReadNotifications.get(0).getNotification());
    }
}
