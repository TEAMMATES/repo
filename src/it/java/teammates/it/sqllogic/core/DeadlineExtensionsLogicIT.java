package teammates.it.sqllogic.core;

import java.time.Instant;

import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import teammates.common.datatransfer.SqlDataBundle;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.common.util.HibernateUtil;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.DeadlineExtensionsLogic;
import teammates.storage.sqlentity.DeadlineExtension;
import teammates.storage.sqlentity.FeedbackSession;
import teammates.storage.sqlentity.Student;

/**
 * SUT: {@link DeadlineExtensionsLogic}.
 */
public class DeadlineExtensionsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private DeadlineExtensionsLogic deadlineExtensionsLogic = DeadlineExtensionsLogic.inst();
    private SqlDataBundle typicalDataBundle;

    @Override
    @BeforeClass
    public void setupClass() {
        super.setupClass();
        typicalDataBundle = getTypicalSqlDataBundle();
    }

    @Override
    @BeforeMethod
    protected void setUp() throws Exception {
        super.setUp();
        persistDataBundle(typicalDataBundle);
        HibernateUtil.flushSession();
        HibernateUtil.clearSession();
    }

    @Test
    public void testGetExtendedDeadline_extensionExists_success() {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student1InCourse1");

        assert student != null;
        Instant extendedDeadlineForStudent = deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student);

        assertNotNull(extendedDeadlineForStudent);
        assertEquals(Instant.parse("2027-04-30T23:00:00Z"), extendedDeadlineForStudent);
    }

    @Test
    public void testGetExtendedDeadline_extensionDoesNotExist_null() {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student2InCourse1");
        Instant extendedDeadlineForStudent = deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student);

        assertNull(extendedDeadlineForStudent);
    }

    @Test
    public void testUpdateDeadlineExtension()
            throws EntityDoesNotExistException, InvalidParametersException {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student1InCourse1");

        DeadlineExtension de = typicalDataBundle.deadlineExtensions.get("student1InCourse1Session1");

        Instant newEndTime = Instant.parse("2028-04-30T23:00:00Z");
        de.setEndTime(newEndTime);
        deadlineExtensionsLogic.updateDeadlineExtension(de);

        assertEquals(newEndTime, deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student));
    }

    @Test
    public void testUpdateDeadlineExtension_invalidParameters_originalUnchanged() {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student1InCourse1");

        DeadlineExtension de = typicalDataBundle.deadlineExtensions.get("student1InCourse1Session1");
        Instant originalEndTime = de.getEndTime();

        Instant invalidEndTime = Instant.parse("2011-04-01T22:00:00Z");
        assert invalidEndTime.isBefore(feedbackSession.getStartTime());
        de.setEndTime(invalidEndTime);

        assertThrows(InvalidParametersException.class, () -> deadlineExtensionsLogic.updateDeadlineExtension(de));

        assertEquals(originalEndTime, deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student));
    }

}
