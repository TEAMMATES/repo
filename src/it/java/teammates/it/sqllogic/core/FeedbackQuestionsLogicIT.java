package teammates.it.sqllogic.core;

import java.util.ArrayList;
import java.util.List;

import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import teammates.common.datatransfer.FeedbackParticipantType;
import teammates.common.datatransfer.SqlDataBundle;
import teammates.common.datatransfer.questions.FeedbackTextQuestionDetails;
import teammates.common.exception.InvalidParametersException;
import teammates.common.util.HibernateUtil;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.FeedbackQuestionsLogic;
import teammates.storage.sqlentity.FeedbackQuestion;
import teammates.storage.sqlentity.FeedbackSession;

/**
 * SUT: {@link FeedbackQuestionsLogic}.
 */
public class FeedbackQuestionsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private FeedbackQuestionsLogic fqLogic = FeedbackQuestionsLogic.inst();

    private SqlDataBundle typicalDataBundle;

    @Override
    @BeforeClass
    public void setupClass() {
        super.setupClass();
        typicalDataBundle = getTypicalSqlDataBundle();
    }

    @Override
    @BeforeMethod
    protected void setUp() throws Exception {
        super.setUp();
        persistDataBundle(typicalDataBundle);
        HibernateUtil.flushSession();
    }

    @Test
    public void testCreateFeedbackQuestion() throws InvalidParametersException {
        FeedbackSession fs = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        FeedbackTextQuestionDetails newQuestionDetails = new FeedbackTextQuestionDetails("New question text.");
        List<FeedbackParticipantType> showTos = new ArrayList<>();
        showTos.add(FeedbackParticipantType.INSTRUCTORS);
        FeedbackQuestion newQuestion = FeedbackQuestion.makeQuestion(fs, 6, "This is a new text question",
                FeedbackParticipantType.STUDENTS, FeedbackParticipantType.OWN_TEAM_MEMBERS, -100,
                showTos, showTos, showTos, newQuestionDetails);

        newQuestion = fqLogic.createFeedbackQuestion(newQuestion);

        FeedbackQuestion actualQuestion = fqLogic.getFeedbackQuestion(newQuestion.getId());

        verifyEquals(newQuestion, actualQuestion);
    }

    @Test
    public void testGetFeedbackQuestionsForSession() {
        FeedbackSession fs = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        FeedbackQuestion fq1 = typicalDataBundle.feedbackQuestions.get("qn1InSession1InCourse1");
        FeedbackQuestion fq2 = typicalDataBundle.feedbackQuestions.get("qn2InSession1InCourse1");
        FeedbackQuestion fq3 = typicalDataBundle.feedbackQuestions.get("qn3InSession1InCourse1");
        FeedbackQuestion fq4 = typicalDataBundle.feedbackQuestions.get("qn4InSession1InCourse1");
        FeedbackQuestion fq5 = typicalDataBundle.feedbackQuestions.get("qn5InSession1InCourse1");
        FeedbackQuestion fq6 = typicalDataBundle.feedbackQuestions.get("qn6InSession1InCourse1NoResponses");

        List<FeedbackQuestion> expectedQuestions = List.of(fq1, fq2, fq3, fq4, fq5, fq6);

        List<FeedbackQuestion> actualQuestions = fqLogic.getFeedbackQuestionsForSession(fs);

        assertEquals(expectedQuestions.size(), actualQuestions.size());
        assertTrue(expectedQuestions.containsAll(actualQuestions));
    }

}
