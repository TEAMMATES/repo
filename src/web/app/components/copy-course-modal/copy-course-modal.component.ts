import {Component, Input, OnInit} from '@angular/core';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
import { COURSE_ID_MAX_LENGTH } from '../../../types/field-validator';
import {FeedbackSession} from "../../../types/api-output";


/**
 * Copy current course modal.
 */
@Component({
  selector: 'tm-copy-course-modal',
  templateUrl: './copy-course-modal.component.html',
  styleUrls: ['./copy-course-modal.component.scss']
})
export class CopyCourseModalComponent implements OnInit {

  // const
  public COURSE_ID_MAX_LENGTH: number = COURSE_ID_MAX_LENGTH;

  public newCourseId : string = '';

  @Input()
  public sessionsInCourse : FeedbackSession[] = [];

  chosenFeedbackSessions: Set<FeedbackSession> = new Set<FeedbackSession>();

  constructor(public activeModal: NgbActiveModal) { }

  ngOnInit(): void {
  }

  /**
   * Fires the copy event.
   */
  copy(): void {
    this.activeModal.close({
      newCourseId: this.newCourseId,
      chosenFeedbackSessionList: Array.from(this.chosenFeedbackSessions)
    });
  }

  /**
   * Toggles selection of course to copy to in set.
   */
  select(session: FeedbackSession): void {
    this.chosenFeedbackSessions.has(session) ? this.chosenFeedbackSessions.delete(session) : this.chosenFeedbackSessions.add(session);
  }

}
