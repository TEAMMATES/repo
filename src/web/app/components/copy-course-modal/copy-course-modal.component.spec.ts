import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
import { CopyCourseModalComponent } from './copy-course-modal.component';
import { By } from '@angular/platform-browser';
// import {
//   Course,
//   FeedbackSession,
//   FeedbackSessionPublishStatus,
//   FeedbackSessionSubmissionStatus,
//   ResponseVisibleSetting,
//   SessionVisibleSetting,
// } from '../../../types/api-output';

describe('CopyCourseModalComponent', () => {
  let component: CopyCourseModalComponent;
  let fixture: ComponentFixture<CopyCourseModalComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ CopyCourseModalComponent ],
      imports: [
        FormsModule,
      ],
      providers: [
        NgbActiveModal,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CopyCourseModalComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  // const CourseToCopy: Course = {
  //   courseId: "Test01",
  //   courseName: "Test course name",
  //   creationTimestamp: 1557764430000,
  //   deletionTimestamp: 0,
  //   timeZone: 'UTC',
  // };
  // const feedbackSessionToCopy: FeedbackSession = {
  //   courseId: 'Test01',
  //   timeZone: 'Asia/Singapore',
  //   feedbackSessionName: 'Test session',
  //   instructions: 'Answer all',
  //   submissionStartTimestamp: 1555232400,
  //   submissionEndTimestamp: 1555332400,
  //   gracePeriod: 0,
  //   sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
  //   responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
  //   submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
  //   publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
  //   isClosingEmailEnabled: true,
  //   isPublishedEmailEnabled: true,
  //   createdAtTimestamp: 1554967204,
  // };

  it('should snap with some course id', () => {
    component.newCourseId = "Test02";
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();

  });

  it('should enable copy button after new courseId is provided', () => {
    component.newCourseId = "Test02";
    fixture.detectChanges();
    const copyButton: any = fixture.debugElement.query(By.css('button.btn.btn-primary'));
    expect(copyButton.nativeElement.disabled).toBeFalsy();
  });

  it('should disable copy if courseId is empty', () => {
    component.newCourseId = "";
    fixture.detectChanges();
    const copyButton: any = fixture.debugElement.query(By.css('button.btn.btn-primary'));
    expect(copyButton.nativeElement.disabled).toBeTruthy();
  });

});
