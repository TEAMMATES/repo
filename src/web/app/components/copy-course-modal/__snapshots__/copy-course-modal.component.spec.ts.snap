// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyCourseModalComponent should snap with default fields 1`] = `
<tm-copy-course-modal
  COURSE_ID_MAX_LENGTH={[Function Number]}
  activeModal={[Function NgbActiveModal]}
  chosenFeedbackSessions={[Function Set]}
  newCourseId=""
  sessionsInCourse={[Function Array]}
>
  <div
    class="modal-header"
  >
    <h5
      class="modal-title"
    >
      <div>
        Copy course and its sessions
      </div>
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
    id="copy-course-modal"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="form-group"
        >
          <label>
            <b>
              Choose new Course ID
            </b>
          </label>
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="copy-course-id"
            maxlength="40"
            type="text"
          />
          <span>
            40 characters left
          </span>
        </div>
        <div
          class="modal-selection"
        >
          <button
            class="btn btn-info btn-sm"
            type="button"
          >
            Select All
          </button>
          <button
            class="btn btn-light btn-sm"
            type="button"
          >
            Clear All
          </button>
        </div>
        
      </div>
    </div>
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      Cancel
    </button>
    <button
      class="btn btn-primary"
      disabled=""
      id="btn-confirm-copy-course"
      type="button"
    >
      Copy
    </button>
  </div>
</tm-copy-course-modal>
`;

exports[`CopyCourseModalComponent should snap with some course id 1`] = `
<tm-copy-course-modal
  COURSE_ID_MAX_LENGTH={[Function Number]}
  activeModal={[Function NgbActiveModal]}
  newCourseId={[Function String]}
>
  <div
    class="modal-header"
  >
    <h5
      class="modal-title"
    >
      <div>
        Copy this course with sessions
      </div>
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
    id="copy-course-modal"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="form-group"
        >
          <label>
            <b>
              Id for copied course
            </b>
          </label>
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="copy-course-id"
            maxlength="40"
            type="text"
          />
          <span>
            34 characters left
          </span>
        </div>
      </div>
    </div>
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      Cancel
    </button>
    <button
      class="btn btn-primary"
      id="btn-confirm-copy-course"
      type="button"
    >
      Copy
    </button>
  </div>
</tm-copy-course-modal>
`;
